src/app/drive/services/folder.service.ts

```diff
@@ -416,34 +446,50 @@ export async function downloadFolderAsZip({
     const { files, token: fileToken } = await addAllSharedFilesToZip(
       folderToDownload.name,
       async (file) => {
-        const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
-        const cachedFile = await lruFilesCacheManager.get(file.uuid?.toString());
-        const isCachedFileOlder = checkIfCachedSourceIsOlder({ cachedFile, file });
-
-        updateNumItems();
-
-        if (cachedFile?.source && !isCachedFileOlder) {
-          updateProgress(1);
-          return cachedFile.source.stream();
+        addUniqueItem(file);
+        // TODO: QA REMOVE
+        if (file.name.includes('internxt_test_file') || file.plainName?.includes('internxt_test_file')) {
+          failedItems.push(file);
+          return;
+        }
+        try {
+          const lruFilesCacheManager = await LRUFilesCacheManager.getInstance();
+          const cachedFile = await lruFilesCacheManager.get(file.uuid?.toString());
+          const isCachedFileOlder = checkIfCachedSourceIsOlder({ cachedFile, file });
+
+          updateNumItems();
+
+          if (cachedFile?.source && !isCachedFileOlder) {
+            updateProgress(1);
+            return cachedFile.source.stream();
+          }
+
+          const downloadedFileStream = await downloadFile({
+            bucketId: file.bucket as string,
+            // TODO: TO WORK UNTIL SDK TYPE CORRECT THE field fileiId -> fileId
+            fileId: (file as any).fileId,
+            creds: options.credentials,
+            mnemonic: options.mnemonic,
+          });
+
+          const sourceBlob = await binaryStreamToBlob(downloadedFileStream);
+          await updateDatabaseFileSourceData({
+            folderId: file.folderId,
+            sourceBlob,
+            fileId: file.id,
+            updatedAt: file.updatedAt,
+          });
+
+          return sourceBlob.stream();
+        } catch (error: any) {
+          const serverUnavailableError = error.message === 'Server unavailable';
+          const isLostConnectionError = error instanceof ConnectionLostError || error.message === 'Network Error';
+          if (serverUnavailableError || isLostConnectionError) {
+            throw error;
+          }
```

Comment:
This check is in different places, extract it to avoid repeating the code and make it easier to maintain and apply the suggested changes